fiaas_deploy_daemon/deployer/kubernetes/deployment/deployer.py

```diff
@@ -151,21 +151,17 @@ def _make_volume_mounts(self, app_spec):
         return volume_mounts
 
     def _make_env(self, app_spec):
-        constants = self._fiaas_env.copy()
-        constants["ARTIFACT_NAME"] = app_spec.name
-        constants["IMAGE"] = app_spec.image
-        constants["VERSION"] = app_spec.version
-        env = [EnvVar(name=name, value=value) for name, value in constants.iteritems()]
-
-        # For backward compatibility. https://github.schibsted.io/finn/fiaas-deploy-daemon/pull/34
-        global_env = []
-        for name, value in self._global_env.iteritems():
-            if "FIAAS_{}".format(name) not in constants and name not in constants:
-                global_env.extend([EnvVar(name=name, value=value), EnvVar(name="FIAAS_{}".format(name), value=value)])
-            else:
-                LOG.warn("Reserved environment-variable: {} declared as global. Ignoring and continuing".format(name))
-        env.extend(global_env)
+        fiaas_managed_env = {
```

Comment:
Contrary to the global envs, which is managed and named by the cluster operator, these variables are fully managed and named by fiaas, and should probably have had the `FIAAS_` prefix from the beginning. That would have avoided the issue where `VERSION` conflicts with something in Nodejs (wasn't it?).
Of course, since it's been like this a while, it might need a deprecation period and proper handling of the migration, so that users are not caught out.

Reply:
Yes, I agree that these variables should have the prefix `FIAAS_`, and like you point out; changing that will require adding copies with the prefix, and a period of having both the current and the prefixed variable keys. Maybe it makes sense to make that change in a separate PR though?

Reply:
Yeah, probably best to do those as separate PRs. 👍